src/parquet/arrow/schema.cc

```diff
@@ -485,14 +485,19 @@ Status FieldToNode(const std::shared_ptr<Field>& field,
     case ArrowType::TIMESTAMP: {
       auto timestamp_type = static_cast<::arrow::TimestampType*>(field->type().get());
       auto unit = timestamp_type->unit();
-      type = ParquetType::INT64;
       if (unit == ::arrow::TimeUnit::MILLI) {
+        type = ParquetType::INT64;
         logical_type = LogicalType::TIMESTAMP_MILLIS;
       } else if (unit == ::arrow::TimeUnit::MICRO) {
+        type = ParquetType::INT64;
         logical_type = LogicalType::TIMESTAMP_MICROS;
+      } else if (unit == ::arrow::TimeUnit::NANO) {
```

Comment:
I was thinking that we'd create an `ArrowWriterProperties` where:

* default is write nanos to int64 with casting to micros
* add option to coerce nanoseconds to millis or micros
* add option to permit unsafe casts (e.g. losing nanoseconds), default to raise if nanoseconds will be dropped

It should be like https://github.com/apache/parquet-cpp/blob/master/src/parquet/column/properties.h, and not break the current API. @c-nichols would you mind starting something simple? @xhochy we can also do this in a follow up patch